components/dialog/dialog-mixin.js

```diff
@@ -177,13 +177,37 @@ export const DialogMixin = superclass => class extends RtlMixin(superclass) {
 		}
 	}
 
+	_findAutofocusElement(node) {
+		if (this._useNative) {
+			// Do not override native autofocus attribute implementation
+			return null;
+		}
+
+		const slot = node.querySelector('slot');
+		if (!slot) {
+			// We are in a confirm dialog, autofocus attribute will never be set
+			return null;
+		}
+
+		const content = slot.assignedElements({ flatten: true });
+
+		const searchOneElement = (el) => {
+			const foundElementOrNull = el.hasAttribute('autofocus') ? el : el.querySelector('[autofocus]');
+			return isFocusable(foundElementOrNull, false, false) ? foundElementOrNull : null;
+		};
+
+		let autofocusElement = null;
+		content.some(x => (autofocusElement = searchOneElement(x)));
```

Comment:
Alternatively, I could do:

```suggestion
		for (const el of content) {
			const foundElement = searchOneElement(el);
			if (foundElement) {
				autofocusElement = foundElement;
				break;
			}
		}
```